config/app-modules.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+return [
+	/*
+	|--------------------------------------------------------------------------
+	| Modules Namespace
+	|--------------------------------------------------------------------------
+	|
+	| This is the PHP namespace that your modules will be created in. For
+	| example, a module called "Helpers" will be placed in \Modules\Helpers
+	| by default.
+	|
+	| It is *highly recommended* that you configure this to your organization
+	| name to make extracting modules to their own package easier (should you
+	| choose to ever do so).
+	|
+	| If you set the namespace, you should also set the vendor name to match.
+	|
+	*/
+	
+	'modules_namespace' => 'Modules',
+	
+	/*
+	|--------------------------------------------------------------------------
+	| Composer "Vendor" Name
+	|--------------------------------------------------------------------------
+	|
+	| This is the prefix used for your composer.json file. This should be the
+	| kebab-case version of your module namespace (if left null, we will
+	| generate the kebab-case version for you).
+	|
+	*/
+	
+	'modules_vendor' => null,
+	
+	/*
+	|--------------------------------------------------------------------------
+	| Modules Directory
+	|--------------------------------------------------------------------------
+	|
+	| If you want to install modules in a custom directory, you can do so here.
+	| Keeping the default `app-modules/` directory is highly recommended,
+	| though, as it keeps your modules near the rest of your application code
+	| in an alpha-sorted directory listing.
+	|
+	*/
+	
+	'modules_directory' => 'app-modules',
```

Comment:
Is it possible to move the modules into the app folder?

Reply:
Moving modules under the app folder causes the following issues:
```php
Error:
Class "Modules\Admin\Providers\AdminServiceProvider" not found

  at /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php:206
  at Illuminate\Foundation\ProviderRepository->createProvider()
     (/var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php:142)
  at Illuminate\Foundation\ProviderRepository->compileManifest()
     (/var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php:61)
  at Illuminate\Foundation\ProviderRepository->load()
     (/var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php:765)
  at Illuminate\Foundation\Application->registerConfiguredProviders()
     (/var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/RegisterProviders.php:17)
  at Illuminate\Foundation\Bootstrap\RegisterProviders->bootstrap()
     (/var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php:263)
  at Illuminate\Foundation\Application->bootstrapWith()
     (/var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:186)
  at Illuminate\Foundation\Http\Kernel->bootstrap()
     (/var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:170)
  at Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter()
     (/var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:144)
  at Illuminate\Foundation\Http\Kernel->handle()
     (/var/www/html/public/index.php:51)
  at require_once('/var/www/html/public/index.php')
     (/var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/resources/server.php:16)        

```
Also it breaks some dependencies. 

To avoid future issues I created a new branch where the modular structure is implemented under the app folder: https://github.com/learnhubmkd/api/tree/laravel-modular

Reply:
Yes, I did that, but doesn't help.

Reply:
In that case, we can use the other branch, feel free to create a pull request from that branch

Reply:
PR opened #31